seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/connection/SimpleJdbcConnectionProvider.java

```diff
@@ -108,8 +108,19 @@ public Connection getOrEstablishConnection() throws SQLException, ClassNotFoundE
         if (jdbcConfig.getPassword().isPresent()) {
             info.setProperty("password", jdbcConfig.getPassword().get());
         }
+        // Apply server time zone for MySQL if user configured it and did not override
+        // serverTimezone
+        String serverTimeZone = jdbcConfig.getServerTimeZone();
+        String url = jdbcConfig.getUrl();
+        if (serverTimeZone != null
+                && !serverTimeZone.isEmpty()
+                && url != null
+                && url.startsWith("jdbc:mysql:")
+                && !jdbcConfig.getProperties().containsKey("serverTimezone")) {
+            info.setProperty("serverTimezone", serverTimeZone);
+        }
         info.putAll(jdbcConfig.getProperties());
-        connection = driver.connect(jdbcConfig.getUrl(), info);
+        connection = driver.connect(url, info);
```

Comment:
I don't think it's suitable to be placed here. You can push the Settings to config